src/Rector/CodeQuality/SimplifyReturnsRector.php

```diff
@@ -0,0 +1,147 @@
+{
+    /**
+     * @var string[]
+     */
+    private $binaryOperations = [
+        Identical::class,
+        NotIdentical::class,
+        Equal::class,
+        NotEqual::class,
+        Greater::class,
+        Smaller::class,
+        GreaterOrEqual::class,
+        SmallerOrEqual::class,
+    ];
+
+    /**
+     * @var BinaryOp
+     */
+    private $condition;
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Symplify if/else conditions that return something.',
+            [new CodeSample(
+                <<<'CODE_SAMPLE'
+if ($something === 1) {
+    return true;
+} else {
+    return false;
+}
+CODE_SAMPLE
+                ,
+                'return $something === 1;'
+            )]
+        );
+    }
+
+    public function isCandidate(Node $node): bool
+    {
+        if (! $node instanceof If_) {
+            return false;
+        }
+
+        $condition = $node->cond;
+
+        if (! in_array(get_class($condition), $this->binaryOperations, true)) {
+            return false;
+        }
+
+        /** @var BinaryOp $binaryOperation */
+        $binaryOperation = $condition;
+
+        $this->condition = $binaryOperation;
+
+        $nextNode = $this->condition->getAttribute(Attribute::NEXT_NODE);
+
+        if (! $nextNode instanceof Return_) {
+            return false;
+        }
+
+        $returnNode = $nextNode;
+
+        $returnExpr = $returnNode->expr;
+
+        if (! $returnExpr instanceof ConstFetch) {
+            return false;
+        }
+
+        $constFetchNode = $returnExpr;
+
+        if ($constFetchNode->name->toLowerString() !== 'true') {
+            return false;
+        }
+
+        $nextNodeAfterReturn = $returnNode->getAttribute(Attribute::NEXT_NODE);
+        $nextNodeAfterIf = $node->getAttribute(Attribute::NEXT_NODE);
+
+        if (! $nextNodeAfterReturn instanceof Else_ && !$nextNodeAfterIf instanceof Return_) {
+            return false;
+        }
+
+        if ($nextNodeAfterIf !== null) {
+            $returnExpr = $nextNodeAfterIf->expr;
+            if (! $returnExpr instanceof ConstFetch) {
+                return false;
+            }
+
+            $constFetchNode = $returnExpr;
+
+            return $constFetchNode->name->toLowerString() === 'false';
+        }
+
+        $nextNode = $nextNodeAfterReturn;
+
+        $elseExpr = $nextNode->stmts[0];
+
+        if (! $elseExpr instanceof Return_) {
+            return false;
+        }
+
+        $returnExpr = $elseExpr->expr;
+
+        if (! $returnExpr instanceof ConstFetch) {
+            return false;
+        }
+
+        $constFetchNode = $returnExpr;
+
+        return $constFetchNode->name->toLowerString() === 'false';
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        $nextPossibleNode = $node->getAttribute(Attribute::NEXT_NODE);
+
+        if ($nextPossibleNode !== null) {
+            // remove next node
```

Comment:
@TomasVotruba My friend, how do we remove next nodes?

Reply:
I know how to remove current node, see:

https://github.com/rectorphp/rector/blob/24abbbe9ad19acde1c23cfcdeb79bffe167828b9/src/Rector/MethodBody/FluentReplaceRector.php#L111

Reply:
Using `$this->removeNode = true; ` didn't work neither for the current node. I'll try to call the `refactor` method, otherwise, we've founded a Rector's limitation :worried:

Reply:
It works for that example, it's tested.

Reply:
@nikic Is even possible to do that? Maybe, on line 145 overwrite the `$attributes` with `['nextNode' => null]`?